build_runner/lib/src/daemon/daemon_builder.dart

```diff
@@ -111,10 +111,18 @@ class BuildRunnerDaemonBuilder implements DaemonBuilder {
               '${target.target}/**', _buildOptions.packageGraph.root.name));
       }
     }
+    Iterable<AssetId>? outputs;
+
     try {
       var mergedChanges = collectChanges([changes]);
       var result = await _builder.run(mergedChanges,
           buildDirs: buildDirs, buildFilters: buildFilters);
+
+      outputs = {
+        for (var change in changes) change.id,
+        ...result.outputs,
+      };
```

Comment:
Ideally we wouldn't create this set either if we don't need it. Probably not a huge deal, but if it isn't too ugly to omit it in that case I would.

Reply:
My idea was to do this in `build_daemon` and to strip the value out in the server so that different implementations don't have to all do the same check. I thought that combining two collections we already have (both `changes` and `result.outputs` are available either way) into a single set was cheap enough to not worry about it, but I can change it if we want to avoid doing it unconditionally. As far as I'm concerned there's only a single build implementation either way :smile:

Reply:
It just depends how easy it is to avoid - it definitely takes some time to allocate a new set and then its going to end up growing it a bunch of times etc as it adds items. So probably a measurable amount of time for larger builds, although dwarfed by overall build time.

If it would make the code really ugly though I would be ok with it.

Reply:
Good point, and it turned out to be relatively easy to avoid (55ff4a3b98b07e2086b40c57fe8d6f2bd7acad22).